qutebrowser/browser/urlmarks.py

```diff
@@ -141,12 +141,33 @@ class QuickmarkManager(UrlMarkManager):
         - removed gets emitted with the name as argument.
     """
 
+    def _quickmark_directory(self):
+        """Return the directory containing the quickmarks.
+
+        For quickmarks are data, return by default the standard data
+        directory. In order not to break the compatibility with an older
+        behavior, if quickmarks are found in the standard configuration
+        directory, return this directory instead.
+        """
+        config_dir = standarddir.config()
+        data_dir = standarddir.data()
+        if config_dir is not None:
+            isinconfigdir = os.path.isfile(
+                os.path.join(config_dir, 'quickmarks')
```

Comment:
Jakub Klinkovský <notifications@github.com> writes:

>> @@ -141,12 +141,33 @@ class QuickmarkManager(UrlMarkManager):
>>          - removed gets emitted with the name as argument.
>>      """
>>  
>> +    def _quickmark_directory(self):
>> +        """Return the directory containing the quickmarks.
>> +
>> +        For quickmarks are data, return by default the standard data
>> +        directory. In order not to break the compatibility with an older
>> +        behavior, if quickmarks are found in the standard configuration
>> +        directory, return this directory instead.
>> +        """
>> +        config_dir = standarddir.config()
>> +        data_dir = standarddir.data()
>> +        if config_dir is not None:
>> +            isinconfigdir = os.path.isfile(
>> +                os.path.join(config_dir, 'quickmarks')
>
> Here you need to assemble the full file path to make the decision, return the directory name and later assemble the file path again, which is somewhat useless. Can't this method return the full path to the `quickmarks` file directly?

I tried to fit as well as possible to the current structure of the
code. IIUC, the construction of LineParser expects a directory and a
file basename, then if I return a file in _quickmark_directory (that
would be renamed _quickmark_file :-)), I will need to split it in
_init_lineparser.

IMHO, both solutions are equally readable and understandable. Plus, for
they don't go in the performance critical path, readability matters most
here.

I don't mind providing _quickmark_file instead of _quickmark_directory,
but I seriously doubt it will be better.

-- 
Konubinix
GPG Key    : 7439106A
Fingerprint: 5993 BE7A DA65 E2D9 06CE  5C36 75D2 3CED 7439 106A